packages/grant-explorer/src/features/projects/hooks/useProject.ts

```diff
@@ -0,0 +1,18 @@
+import useSWR from "swr";
+import { DataLayer } from "data-layer";
+import { AlloVersion } from "data-layer/src/data-layer.types";
+
+type Params = {
+  projectId?: string;
+};
+
+export function useProject(params: Params, dataLayer: DataLayer) {
+  const shouldFetch = Object.values(params).every(Boolean);
```

Comment:
if projectId is undefined it should return earlier null or undefined, the same returned from the data layer if the project is not found, so that we don't need to cast it later with `as string`